test/integration/controller_test.go

```diff
@@ -97,12 +98,7 @@ func TestBasicFlows(t *testing.T) {
 		},
 		ProvisionReaction: &fakeosb.ProvisionReaction{
 			Response: &osb.ProvisionResponse{
-				Async: true,
-			},
-		},
-		PollLastOperationReaction: &fakeosb.PollLastOperationReaction{
-			Response: &osb.LastOperationResponse{
-				State: osb.StateSucceeded,
+				Async: false,
```

Comment:
I didn't really delete a test case - just made the integration test use a synchronous provision for now.  There's a bug (#780) where I put a comment describing the issue.  TLDR: if you do async provision here, it exposes race conditions that make the test super flaky.  I am not sure at all why this PR happened to expose it - but I verified that the races I was seeing in CI for this PR were due to races between the polling queue worker and the normal instance queue worker.

I'm happy to break this test change out into a separate PR if you'd like.